src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,630 @@
+        }
+
+        private fun handelSkyMall(lore: List<String>) {
+            if (!SKY_MALL.enabled || !SKY_MALL.isUnlocked) HotmAPI.skymall = null
+            else {
+                val index = lore.indexOf("§aYour Current Effect") + 1
+                skymallPattern.matchMatcher(lore[index]) {
+                    val perk = group("perk")
+                    HotmAPI.skymall = SkymallPerk.entries.find { it.itemString == perk }
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onScoreboardUpdate(event: ScoreboardChangeEvent) {
+            if (!LorenzUtils.inSkyBlock) return
+
+            event.newList.matchFirst(ScoreboardPattern.powderPattern) {
+                val type = HotmAPI.Powder.entries.firstOrNull { it.lowName == group("type") } ?: return
+                val amount = group("amount").formatLong()
+                val difference = amount - type.getCurrent()
+
+                if (difference > 0) {
+                    type.gain(difference)
+                    ChatUtils.debug("Gained §a${difference.addSeparators()} §e${type.lowName} Powder")
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onInventoryClose(event: InventoryCloseEvent) {
+            if (!inInventory) return
+            inInventory = false
+            entries.forEach { it.slot = null }
+            heartItem = null
+        }
+
+        @SubscribeEvent
+        fun onInventoryFullyOpen(event: InventoryFullyOpenedEvent) {
+            if (!LorenzUtils.inSkyBlock) return
+            inInventory = inventoryPattern.matches(event.inventoryName)
+            DelayedRun.runNextTick {
+                InventoryUtils.getItemsInOpenChest().forEach { it.parse() }
+                abilities.filter { it.isUnlocked }.forEach {
+                    it.activeLevel = if (PEAK_OF_THE_MOUNTAIN.rawLevel >= 1) 2 else 1
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onChat(event: LorenzChatEvent) {
+            if (!LorenzUtils.inSkyBlock) return
+            if (resetChatPattern.matches(event.message)) {
+                resetTree()
+                return
+            }
+            skymallPattern.matchMatcher(event.message) {
+                val perk = group("perk")
+                HotmAPI.skymall = SkymallPerk.entries.find { it.chat == perk }
+                ChatUtils.debug("setting skymall to ${HotmAPI.skymall}")
+                return
+            }
+            DelayedRun.runNextTick {
+                mayhemChatPattern.matchMatcher(event.message) {
+                    val perk = group("perk")
+                    HotmAPI.mineshaftMayhem = MayhemPerk.entries.find { it.chat == perk }
+                    ChatUtils.debug("setting mineshaftMayhem to ${HotmAPI.mineshaftMayhem}")
+                }
+            }
+        }
+
+        @SubscribeEvent
+        fun onWorldSwitch(event: IslandChangeEvent) {
+            if (HotmAPI.mineshaftMayhem != null) return
```

Comment:
this is broken, i was gonna fix it but you merged it too fast
```suggestion
            if (HotmAPI.mineshaftMayhem == null) return
```